generated/java/google-cloud-pubsub-v1/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberApi.java

```diff
@@ -304,13 +307,11 @@ public final SubscriberSettings getSettings() {
    * @throws com.google.api.gax.grpc.ApiException if the remote call fails
    */
   public final Subscription createSubscription(
-      String name, String topic, PushConfig pushConfig, int ackDeadlineSeconds) {
-    SUBSCRIPTION_PATH_TEMPLATE.validate(name, "createSubscription");
-    TOPIC_PATH_TEMPLATE.validate(topic, "createSubscription");
+      SubscriptionName name, TopicNameOneof topic, PushConfig pushConfig, int ackDeadlineSeconds) {
```

Comment:
topic should be configured to only take TopicName, since a deleted topic is not valid at creation time.